portal/src/util/misc.ts

```diff
@@ -0,0 +1,25 @@
+export function getListFromCommaSeparatedString(
```

Comment:
The behavior of this function seems quite complex. Can you add some tests to demonstrate the intended behavior?